### PR TITLE
AY: creating btrfs subvolumes

### PR DESCRIPTION
**Pull request by schubi2**

### CODE REVIEW

File: src/include/users/dialogs.rb [hunk 1]
```diff
@@ -1921,6 +1924,8 @@ def valid_btrfs_path?(path)
     #
     # @return [Boolean] true if a Btrfs filesystem is found; false otherwise
     def btrfs_available?
+      return true if Mode.config # AutoYast configuration mode
+
```
[the remark this answers is not in the document]
Comment (schubi2) [reply src/include/users/dialogs.rb]: Yes, I have adapted.

[the remark this answers is not in the document]
Comment (dgdavid) [reply src/include/users/dialogs.rb]: Nice!

But remember to keep `valid_btrfs_path?` and `btrfs_available?` as they were. I mean, to remove those

```ruby
 return true if Mode.config # AutoYast configuration mode
```
lines from them.

[the remark this answers is not in the document]
Comment (schubi2) [reply src/include/users/dialogs.rb]: Yes, you are right. It is not needed anymore.

File: testsuite/tests/ImportExportUser.out [hunk 2]
```diff
@@ -2,6 +2,6 @@ Dump	==========================================================
 Dump	importing:
 Dump	 $["encrypted":true, "forename":"aa", "gid":500, "grouplist":"audio", "home":"/home/aaa", "password_settings":$["inact":-1, "max":99999, "min":0, "warn":7], "shell":"/bin/bash", "surname":"A", "uid":500, "user_password":"5OkhN2nLxMfEY", "username":"aaa"]
 Dump	==== processing ImportUser... ============================
-Return	$["authorized_keys":[], "cn":"aa A", "encrypted":true, "gidNumber":"500", "grouplist":$["audio":"1"], "homeDirectory":"/home/aaa", "loginShell":"/bin/bash", "modified":"imported", "shadowInactive":"-1", "shadowLastChange":"0", "shadowMax":"99999", "shadowMin":"0", "shadowWarning":"7", "type":"local", "uid":"aaa", "uidNumber":"500", "userPassword":"5OkhN2nLxMfEY"]
```
Comment (imobachgs) [inline testsuite/tests/ImportExportUser.out]: AFAIS, we import `btrfs_subvolume` and export `home_btrfs_subvolume`?

File: src/modules/Users.pm [hunk 3]
```diff
@@ -6453,6 +6457,10 @@ sub ExportUser {
     if ($user->{"homeDirectory"} && $user->{"authorized_keys"}) {
             $ret{"authorized_keys"} = $user->{"authorized_keys"};
     }
+    my $btrfs_subvolume	= bool ($exported_user{"btrfs_subvolume"});
+    if (defined $btrfs_subvolume) {
+	$ret{"home_btrfs_subvolume"} = YaST::YCP::Boolean ($btrfs_subvolume);
```
Comment (imobachgs) [inline src/modules/Users.pm]: Why do we need to rename this value?

File: testsuite/tests/ImportExportUser.out (hunk 2 above)
Comment (schubi2) [reply testsuite/tests/ImportExportUser.out]: Yes, I know "btrfs_subvolume" is the internal variable. I have used "home_btrfs_subvolume" for the AutoYaST file. It is more userfriendly.

**Comment by imobachgs** [comment]
Missing bug number was fixed in #199 .

**Pull request merged**